### PR TITLE
[source-postgres] Feat: Use latest CDK for new getLsn()

### LINKED ISSUE #45396: [source-postgres] Logical replication not working on read replicas
### Connector Name

source-postgres

### Connector Version

 3.6.18

### What step the error happened?

Syncs never start.

### Relevant information

CDC Replication doesn't work on Postgres read replicas because it uses functions that can only be executed on the master - probably only one function:  `pg_current_wal_lsn()`. 
I believe this can be solved quickly by replacing usage of `pg_current_wal_lsn` function with `pg_last_wal_receive_lsn`.

Postgres version : 16.13R2

Good to know: [Debezium has solved this issue on their side very simply since V2.5](https://debezium.io/blog/2023/12/04/debezium-2-5-beta1-released/#:~:text=Streaming%20from%20PostgreSQL%2016%20stand%2Dbys), however the connector doesn't use Debezium for everything. 

### Relevant log output

```shell
java.lang.RuntimeException: org.postgresql.util.PSQLException: ERROR: recovery is in progress Hint: WAL control functions cannot be executed during recovery.
```

### Contribute

- [X] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by assaadhjb**
## What
Using latest Airbyte Java CDK version
This follows https://github.com/airbytehq/airbyte/pull/45397 and resolves https://github.com/airbytehq/airbyte/issues/45396
## How
Using the new Airbyte Java CDK version, destination Postgres is able to use a `getLsn()` function that works on both read-replicas and master.

## User Impact
Users will be able to replicate from Postgres read-replicas for PG 16 onwards.

## Can this PR be safely reverted and rolled back?

- [X] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by theyueli** [comment]
please fix failed CI (compilation error)

**Comment by assaadhjb** [comment]
@theyueli airbyte Java CDK 0.47.1 could not be found, after https://github.com/airbytehq/airbyte/pull/45397 I pushed a fix here https://github.com/airbytehq/airbyte/pull/46101. So maybe we'd need to redeploy?

**Comment by theyueli** [comment]
> @theyueli airbyte Java CDK 0.47.1 could not be found, after #45397 I pushed a fix here #46101. So maybe we'd need to redeploy?

I think the reason is that the CDK was not published. I was trying to publish the CDK but since your PR is a community PR, our github action didn't work out ... I'm checking with @marcosmarxm to see what's the right way to do it.

**Comment by theyueli** [comment]
> > @theyueli airbyte Java CDK 0.47.1 could not be found, after #45397 I pushed a fix here #46101. So maybe we'd need to redeploy?
> 
> I think the reason is that the CDK was not published. I was trying to publish the CDK but since your PR is a community PR, our github action didn't work out ... I'm checking with @marcosmarxm to see what's the right way to do it.

The CDK now should be available. Could you try again to see if it compiles?

File: docs/integrations/sources/postgres.md [hunk 1]
```diff
@@ -329,6 +329,7 @@ According to Postgres [documentation](https://www.postgresql.org/docs/14/datatyp
 
 | Version | Date       | Pull Request                                             | Subject                                                                                                                                                                   |
 |---------|------------|----------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 3.6.19  | 2024-09-17 | - | Adopt latest CDK to allow cascading replication (Postgres > 16)                 |
```
Comment (theyueli) [inline docs/integrations/sources/postgres.md:332]: This needs to be 3.6.20

**Review by theyueli — APPROVED** [review]

**Comment by assaadhjb** [comment]
@theyueli some globals seem to have been moved since last CDK version, hence the CI issue, I'll try fix it

**Comment by theyueli** [comment]
hi @assaadhjb , 

The compilation error is introduced due to recent refactoring of the CDK, which maybe more difficult for you to resolve without knowing the background. 

I just merged a PR to address all of these. 
https://github.com/airbytehq/airbyte/pull/46299

Therefore, this PR's purpose has been achieved above. I will close this one as it is completed. 

You can now test your contribution using the latest 3.6.20 postgres source connector now. Please keep us posted. Thanks again for your contribution!

**Pull request closed without merging**